### PR TITLE
Feature/swagger

### PR DESCRIPTION
**Pull request by Di3GO95**
Swagger 2 implementation through anotations.

### CODE REVIEW

File: src/main/java/org/diego/tutorial/car/databases/jpa/JPAImpl.java [hunk 1]
```diff
@@ -40,7 +41,7 @@
 		try {
 			em.persist(entity);
 		} catch (PersistenceException e) {
-			throw new DataNotFoundException("Trying to add an object that already exists");
+			throw new DataAlreadyExistsException("Trying to add an object that already exists");
```
[the remark this answers is not in the document]
Comment (Di3GO95) [reply src/main/java/org/diego/tutorial/car/databases/jpa/JPAImpl.java]: em.persist throws an EntityExistsException:

If the entity already exists, the EntityExistsException maybe thrown when the persist operation is invoked, or the EntityExistsException or another PersistenceException may bethrown at flush or commit time.

So I thought that I should include those exceptions in the more general exception, PersistenceException, and then throw a DataNotFoundException.

[the remark this answers is not in the document]
Comment (javieraviles) [reply src/main/java/org/diego/tutorial/car/databases/jpa/JPAImpl.java]: how can a car already exist in the DB? such condition would only happen if ID is already in DB and that is created within the backend (incrementally?), not provided through the API, so you cannot really tell the user that the car is already in db unless I'm really missing something here

[the remark this answers is not in the document]
Comment (Di3GO95) [reply src/main/java/org/diego/tutorial/car/databases/jpa/JPAImpl.java]: True, it's impossible. I am going to update it.

**Comment by Di3GO95** [comment]
> Binaries should not be in the repo either

Gitignore updated in develop branch, and merged in this and other branches, where _bin_ folder is excluded

**Pull request merged**